pendulum_control/CMakeLists.txt

```diff
@@ -0,0 +1,60 @@
+cmake_minimum_required(VERSION 2.8.3)
+
+project(pendulum_control)
+
+if(NOT WIN32)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -Wextra")
+endif()
+
+find_package(ament_cmake REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(rmw_implementation REQUIRED)
+find_package(pendulum_msgs REQUIRED)
+find_package(rttest)
+if (NOT rttest_FOUND)
+  message(ERROR "rttest not found. pendulum_control package will not be built.")
+  return()
+endif()
+
+get_available_rmw_implementations(middleware_implementations)
+foreach(middleware_impl ${middleware_implementations})
+  # Find package each of them.
+  find_package("${middleware_impl}" REQUIRED)
+endforeach()
+
+function(add_executable_for_each_rmw_impl executable)
+  # Build the executable for the default rmw implementation.
+  add_executable(${executable} ${ARGN})
+  # TODO improve inclusion of ncurses
+  target_link_libraries(${executable})
+  ament_target_dependencies(${executable}
+    "rclcpp"
+    "rmw_implementation"
+    "pendulum_msgs"
+    "rttest"
+  )
+
+  install(TARGETS ${executable} DESTINATION bin)
+
+  # Build an executable for each rmw implementation.
+  foreach(middleware_impl_tmp ${middleware_implementations})
+    add_executable(${executable}__${middleware_impl_tmp} ${ARGN})
+    target_link_libraries(${executable}__${middleware_impl_tmp})
+    ament_target_dependencies(${executable}__${middleware_impl_tmp}
+      "rclcpp"
+      "${middleware_impl_tmp}"
+      "pendulum_msgs"
+      "rttest"
+    )
+
+    install(TARGETS ${executable}__${middleware_impl_tmp} DESTINATION bin)
+  endforeach()
+
+endfunction()
```

Comment:
ros2/rmw_implementation#3 adds the function to `rmw_implementation`.

In the following PRs you see how it is being used: ros2/examples#48 ros2/demos#10